### ISSUE TITLE
Conditionally copy a file

### ISSUE THREAD

**Issue by ark-** [ffizer/ffizer#263]
I have binary files which completely different content which I want to copy only if the user selects a variable.
e.g. .ffizer.yaml
````
  - name: BIN_TYPE
    default_value: fast
````
In my template directory I have 3 files

 * `fast.bin`
 * `slow.bin`
 * `balanced.bin`

I would like to only render the selected binary type. 

In the README, the following phrase makes me think this must be possible

````
file / folder ignored under conditions (ignores'item in ffizer.yaml are defined as handlerbar expression)
````

So I have tried the following (and multiple variants of it) to no prevail. Always says it's not a valid exclude statement.

````
ignores:
  - .git # exclude .git of the template host
  - {{#unless (eq BIN_TYPE 'fast')}}**/fast.bin{{/unless}}
````

**Comment by davidB** [comment]
yaml doesn't like `{{` as is try with quotes
```
ignores:
  - .git # exclude .git of the template host
  - '{{#unless (eq BIN_TYPE "fast")}}**/fast.bin{{/unless}}'
```

 (I didn't know about `unless` block)

**Comment by ark-** [comment]
Thanks I'll try this. I tried with double quotes but didn't try single.  Will report back.

**Comment by ark-** [comment]
Still doesn't like the string even in single quotes. See error below:

````
Jun 01 09: 13: 21.377 ERRO failed, 
cmd: CliOpts { 
    verbose: 0, 
    cmd: Apply(ApplyOpts { 
        confirm: Never, 
        update_mode: Ask, 
        no_interaction: false, 
        offline: false, 
        src: SourceLoc { 
            uri: SourceUri { 
                raw: "run-template", 
                path: "run-template", 
                host: None
            }, 
            rev: "master", 
            subfolder: None
        }, 
        dst_folder: "test-output"
    })
}, error: SerdeYaml { 
    source: Message("invalid value: string \"{{#unless (eq BIN_TYPE \\\"fast\\\")}}**/fast.bin{{/unless}}\", expected valid path matcher", 
        Some(Pos { 
            marker: Marker { index: 813, line: 29, col: 4
        }, 
        path: "ignores[1]"
    }))
}
````

**Comment by davidB** [comment]
I made a crappy fix, but I'm refactoring the way to read `.ffizer.yaml ` to fix it in a cleaner way (+ some other future extension). I'll do a release 2.0.0.betaX later this week including the fix.

**Comment by ark-** [comment]
Nice, thank you! Unfortunately your current fix isn't working for me but I'll keep an eye on this issue until you do a proper refactor on the .ffizer.yaml. No point fixing a quickfix when you're going to change things anyway.

**Comment by davidB** [comment]
I released with the fix (and the refactor) 2.0.0-beta.3. You can see the sample at https://github.com/ffizer/ffizer/blob/master/tests/data/test_7_ignores/template/.ffizer.yaml

The correct syntax for your case should be:

```
ignores:
  - .git # exclude .git of the template host
  - "{{#unless (eq BIN_TYPE 'fast')}}**/fast.bin{{/unless}}"
```

**Comment by ark-** [comment]
I've only recently had a chance to check this fix and it does work. If you want to keep this comment open until you do your planned refactor that's OK but feel free to close it too.

**Comment by davidB** [comment]
Thanks for the feedback

**Comment by davidB** [comment]
Reopen because the root cause of the issue was not fixed.

To reproduced, update the sample 7 with: 
```yaml
variables:
  - name: v
    default_value: fast
ignores:
  - .git # exclude .git of the template host
  # - '{{#unless (eq v "fast")}}file3.txt{{/unles}}' #do not work
  - '{{#if (eq v "fast")}}file2.txt{{/if}}'
  - '{{#if (ne v "fast")}}file1.txt{{/if}}'
  - '{{#if (ne v "slow")}}dir1/file4.txt{{/if}}'
  - "{{v}}"
```
with layout 

```txt
tests/data/template_7_ignores
├── .ffizer.samples.d
│   └── my-project.expected
│       ├── dir1
│       │   └── file5.txt
│       ├── file1.txt
│       └── file3.txt
├── .ffizer.yaml
├── dir1
│   ├── file4.txt
│   └── file5.txt
├── fast
├── file1.txt
├── file2.txt
└── file3.txt

4 directories, 10 files
```

and run `cargo run -- test-samples -s tests/data/template_7_ignores`

**Comment by davidB** [comment]
fixed with ffizer 2.1+
